homeassistant/components/suez_water/coordinator.py

```diff
@@ -48,61 +29,22 @@ def __init__(self, hass: HomeAssistant, config_entry: ConfigEntry) -> None:
         )
 
     async def _async_setup(self) -> None:
-        self._sync_client = await self.hass.async_add_executor_job(self._get_client)
+        self._suez_client = SuezClient(
+            username=self.config_entry.data[CONF_USERNAME],
+            password=self.config_entry.data[CONF_PASSWORD],
+            counter_id=self.config_entry.data[CONF_COUNTER_ID],
+        )
+        if not await self._suez_client.check_credentials():
+            raise ConfigEntryError("Invalid credentials for suez water")
```

Comment:
```suggestion
        try:
            self._suez_client = SuezClient(
                username=self.config_entry.data[CONF_USERNAME],
                password=self.config_entry.data[CONF_PASSWORD],
                counter_id=self.config_entry.data[CONF_COUNTER_ID],
            )
            if not await self._suez_client.check_credentials():
                raise ConfigEntryError("Invalid credentials for suez water")
        except PySuezError as ex:
            raise UpdateFailed from ex
```

Reply:
I don't think this is needed as no exception can be thrown from the method.
Or only a connexion error but nothing handled from the lib

![image](https://github.com/user-attachments/assets/08ca6328-b324-4033-af98-1f048465cff5)

Reply:
Oh in that case this is fine